fmriprep/workflows/bold/base.py

```diff
@@ -539,12 +536,11 @@ def init_func_preproc_wf(bold_file):
                 ('outputnode.bold', 'bold_files')]),
             (join_echos, final_boldref_wf, [
                 ('bold_files', 'inputnode.bold_file')]),
-            (bold_bold_trans_wf, skullstrip_bold_wf, [
-                ('outputnode.bold', 'inputnode.in_file')]),
-            (skullstrip_bold_wf, join_echos, [
-                ('outputnode.skull_stripped_file', 'skullstripped_bold_files')]),
+            # use reference image mask used by bold_bold_trans_wf
+            (bold_bold_trans_wf, bold_t2s_wf, [
```

Comment:
I'd be careful to ensure that `bold_bold_trans_wf` does not depend on `bold_t2s_wf`. If this assumption is acceptable, then this seems the most reasonable solution to this problem.

Reply:
Is that because it would introduce a loop in the workflow? I agree that T2* mapping/optimal combination should happen after the native-space transforms are applied, mostly so that those transforms can be done all at once. There are solid reasons to do optimal combination after distortion correction, but I don't think that will have a very large effect on the data and I doubt that it's worth the extra effort since it would involve splitting up the transforms performed by `bold_bold_trans_wf` into two sets.

Reply:
> Is that because it would introduce a loop in the workflow?

Yes, Nipype will not handle loops.